extensions/ARM/SPV_ARM_tensors.asciidoc

```diff
@@ -86,8 +86,18 @@ Modify the definition of the following terms in 2.2.2 Types:
 
 'Composite Type': Add **OpTypeTensorARM** whose 'Shape' operand is present to the list of composite types.
 
-Capabilities
-~~~~~~~~~~~~
+Decoration
+~~~~~~~~~~
+
+Modify Section 3.20, "Decoration", adding
+
+"A tensor variable in the *UniformConstant* storage class (see **OpTypeTensorARM**)"
+
+to the list of cases in which a memory object declaration may be decorated with
+*NonWritable* or *NonReadable*.
```

Comment:
Sorry for the potentially silly question - I feel like I must be missing something here.

If the tensor variable is in the **UniformConstant** storage class isn't it already read-only (and hence **NonWritable**?  Additionally, what would it mean to have a read-only variable that is **NonReadable**?

Reply:
Similarly to what happens for images, the variable in the **UniformConstant** storage class just is a handle to the tensor data. The handle itself is always read-only but the data it points to can be read-write, read-only, or write-only. I agree it is a little bit confusing. You can have a look at the validator changes in https://github.com/KhronosGroup/SPIRV-Tools/pull/6184 to observe the symmetry with images.

Reply:
Got it - thanks!